django/test/testcases.py

```diff
@@ -303,7 +303,10 @@ def assertRedirects(self, response, expected_url, status_code=302,
                 path = urljoin(response.request['PATH_INFO'], path)
 
             if fetch_redirect_response:
-                if netloc:
+                # netloc might be empty, or in cases where Django tests the HTTP scheme,
```

Comment:
Not if you're using a client subclass with an overridden default `SERVER_NAME` which is a common pattern when testing projects using `django-hosts`. I suggest we use `response.client.defaults.get('SERVER_NAME', 'testclient')`.